e2e/settings/settings.steps.ts

```diff
@@ -0,0 +1,49 @@
+import {binding, then, when} from 'cucumber-tsflow';
+import {World, P, expect} from '../common/world';
+import {OAuthSettingsComponent, SettingsPage} from './settings.po';
+
+@binding([World])
+class SettingsSteps {
+  constructor(protected world: World) {
+  }
+
+  @then(/^"([^"]*)" is presented with "([^"]*)" settings tab$/)
+  public activeTab(user: string, tabName: string): P<any> {
+
+    const page = new SettingsPage();
+    const activeTab = page.activeTab();
+
+    return expect(activeTab)
+      .to.eventually.be.equal(tabName, `Default settings tab should be ${tabName}`);
+  }
+
+  @then(/^settings item "([^"]*)" has button "([^"]*)"$/)
+  public settingsItemHasButton(itemTitle: string, buttonTitle: string): P<any> {
+    const page = new SettingsPage();
+
+    return page.getSettingsItem(itemTitle);
+  }
+
+  @when(/^"([^"]*)" clicks to the "([^"]*)" item "([^"]*)" button$/)
+  public clickSettingsButton(userAlias: string, itemTitle: string, buttonTitle: string): P<any> {
+    const page = new SettingsPage();
+    return page.clickButton(itemTitle, buttonTitle);
+  }
+
+  @when(/^fill form in "([^"]*)" settings item$/)
+  public fillSettingsItemForm(itemTitle: string): P<any> {
+    const settings = new OAuthSettingsComponent();
+    const toFill = this.world.testConfig.settings[itemTitle];
```

Comment:
This test requires update to test settings config map.
I already did that on syndesis-ci.